bot/core/event_middleware.py

```diff
@@ -2,33 +2,25 @@
 
 import typing
 
-import attrs
 import hikari
 
-BAD_ATTRS = ["app", "shard"]
+from .lua.executor import execute_lua
+from .lua.lua_py_dict import LuaPyDict
+from .serialize import serialize
 
 
 def _middleware(
     function: typing.Callable[[hikari.Event], typing.Dict[str, typing.Any]]
-) -> typing.Callable[[hikari.Event], typing.Awaitable[None]]:
-    async def inner(event: hikari.Event) -> None:
+) -> typing.Callable[[hikari.Event], None]:
+    def inner(event: hikari.Event) -> None:
         data = function(event)
-        # Pretend this is the event handler
-        print(data)
+        # pretend this is the executor
+        code = event.message.content  # type: ignore
+        execute_lua(event.app, event.guild_id, code, LuaPyDict(data))  # type: ignore
```

Comment:
This code is a hack and is specific to the on_message event (it's only temporary - we're not going to be evaluating the message content as code, we'll be querying the database). The function sig is correct, but this is safe for now because we only have the on_message event for now.